MediaGalleryRenditions/Model/GetRenditionPath.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryRenditions\Model;
+
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Exception\FileSystemException;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Filesystem;
+use Magento\Framework\Filesystem\DriverInterface;
+use Magento\MediaGalleryApi\Api\Data\AssetInterface;
+use Magento\MediaGalleryRenditionsApi\Api\GetRenditionPathInterface;
+
+class GetRenditionPath implements GetRenditionPathInterface
+{
+    /**
+     * @var Filesystem\Directory\WriteInterface
+     */
+    private $directory;
+
+    /**
+     * @var DriverInterface|mixed|null
+     */
+    private $file;
+
+    private const RENDITIONS_DIRECTORY_NAME = '.renditions';
+
+    /**
+     * @param Filesystem $filesystem
+     * @param DriverInterface|null $file
+     * @throws FileSystemException
+     */
+    public function __construct(
+        Filesystem $filesystem,
+        DriverInterface $file = null
+    ) {
+        $this->directory = $filesystem->getDirectoryWrite(DirectoryList::MEDIA);
```

Comment:
Write access is not required
```suggestion
        $this->directory = $filesystem->getDirectoryRead(DirectoryList::MEDIA);
```

Reply:
Write access is required to create the rendition directory path

Reply:
This class is responsible for constructing and returning a rendition path only, I don't think it should not create directories

Reply:
So how we will handle the issue if the rendition directory is not created in the media directory?

Reply:
My point is that `GetRenditionPath` path class does not create a directory anyways, so the write access is not needed. It will be needed during the operation of saving the rendition to the filesystem